### ISSUE TITLE
Viewing large images is not a good experience on Mac

### ISSUE THREAD

**Issue by mstange** [servo/webrender#2823]
If you go to https://firefoxux.github.io/people/smartell/quantum/firefox-quantum_launch-poster_24x36.jpg and zoom in, and then scroll around on the image, things are very slow and janky, apparently due to texture upload.

**Comment by nical** [comment]
Also not great on linux.

**Comment by kvark** [comment]
Yeah, it appears to only been junky for the timeline of texture being uploaded. When the uploading finishes, it becomes fast again.

**Comment by kvark** [comment]
I took a small [profile](https://perfht.ml/2K1OsrX) to investigate, and it appears that both WR Render backend and Renderer are 90% waiting. Texture uploader in particular only takes 1% of time. Is anything wonky happening on the Gecko side, @aosmond ?

**Comment by aosmond** [comment]
I'm hoping to avoid the extra uploading with https://bugzilla.mozilla.org/show_bug.cgi?id=1469964. We'll see if that resolves the jank or if more is required.
